aws/table_aws_rds_db_proxy.go

```diff
@@ -0,0 +1,243 @@
+				Description: "Provides a list of VPC security groups that the proxy belongs to.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "vpc_subnet_ids",
+				Description: "The EC2 subnet IDs for the proxy.",
+				Type:        proto.ColumnType_JSON,
+			},
+			// Standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DBProxyName"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getRDSDBProxyTags,
+				Transform:   transform.FromValue(),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("DBProxyArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listRDSDBProxies(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create Session
+	svc, err := RDSClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_rds_db_proxy.listRDSDBProxies", "connection_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 20 {
+				maxLimit = 20
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &rds.DescribeDBProxiesInput{
+		MaxRecords: aws.Int32(maxLimit),
+	}
+
+	paginator := rds.NewDescribeDBProxiesPaginator(svc, input, func(o *rds.DescribeDBProxiesPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_rds_db_proxy.listRDSDBProxies", "api_error", err)
+			return nil, err
+		}
+
+		for _, items := range output.DBProxies {
+			d.StreamListItem(ctx, items)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func getRDSDBProxy(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	dbProxyName := d.KeyColumnQuals["db_proxy_name"].GetStringValue()
+
+	if dbProxyName == "" {
+		return nil, nil
+	}
+
+	// Create service
+	svc, err := RDSClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_rds_db_proxy.getRDSDBProxy", "connection_error", err)
+		return nil, err
+	}
+
+	params := &rds.DescribeDBProxiesInput{
+		DBProxyName: aws.String(dbProxyName),
+	}
+
+	op, err := svc.DescribeDBProxies(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_rds_db_proxy.getRDSDBProxy", "api_error", err)
+		return nil, err
+	}
+
+	if op.DBProxies != nil && len(op.DBProxies) > 0 {
```

Comment:
```suggestion
	if  len(op.DBProxies) > 0 {
```
What kind of output do we get in the event of API returning no result? Is it nil or an empty array?

Reply:
As it is a get call we are handling the not found error in our get config. In that case, we are getting a null value by ignoring that error. We can remove the condition since it won't execute in case of not found error.